crypto/evp_extra/p_pqdsa_asn1.c

```diff
@@ -158,31 +158,20 @@ static int pqdsa_priv_decode(EVP_PKEY *out, CBS *params, CBS *key, CBS *pubkey)
 
   // Try to parse as one of the three ASN.1 formats defined in ML-DSA-XX-PrivateKey
   // Currently only the following cases are supported:
-  // Case 1: seed [0] OCTET STRING
-  // Case 2: expandedKey OCTET STRING
+  // Case 1: seed OCTET STRING
+  // Case 2: expandedKey [1] OCTET STRING
 
   // Once https://datatracker.ietf.org/doc/draft-ietf-lamps-dilithium-certificates/
   // is stable we will implement:
   // Case 3: both SEQUENCE { seed, expandedKey }
 
-  if (CBS_peek_asn1_tag(key, CBS_ASN1_CONTEXT_SPECIFIC | 0)) {
-    // Case 1: seed [0] OCTET STRING
-    CBS seed;
-    if (!CBS_get_asn1(key, &seed, CBS_ASN1_CONTEXT_SPECIFIC | 0)) {
-      OPENSSL_PUT_ERROR(EVP, EVP_R_DECODE_ERROR);
-      return 0;
-    }
-
-    if (CBS_len(&seed) != out->pkey.pqdsa_key->pqdsa->keygen_seed_len) {
-      OPENSSL_PUT_ERROR(EVP, EVP_R_INVALID_BUFFER_SIZE);
-      return 0;
-    }
-
-    return PQDSA_KEY_set_raw_keypair_from_seed(out->pkey.pqdsa_key, &seed);
-  } else if (CBS_peek_asn1_tag(key, CBS_ASN1_OCTETSTRING)) {
-    // Case 2: expandedKey OCTET STRING
+  if (CBS_len(key) == out->pkey.pqdsa_key->pqdsa->keygen_seed_len) {
```

Comment:
This should still need to peek to see that it is a `CBS_ASN1_OCTETSTRING` type and then specifically get that asn1 type.

Reply:
[7e68294](https://github.com/aws/aws-lc/pull/2194/commits/7e68294a918491fb59e8d8a48635298e504f934e)